rasterio/_warp.pyx

```diff
@@ -585,3 +488,83 @@ def _calculate_default_transform(
     dst_height = nlines
 
     return dst_affine, dst_width, dst_height
+
+cdef void *_create_MEM_dataset(data, transform, crs, desc, src_count=None) except *:
```

Comment:
No, just my lack of knowledge of it. :)  If that could replace this function, that's great - 💯 for code reuse!